cardano-node/src/Cardano/Tracing/ConvertTxId.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Cardano.Tracing.ConvertTxId
+  ( ConvertTxId (..)
+  ) where
+
+import           Cardano.Prelude hiding (All)
+
+import           Data.SOP.Strict
+
+import qualified Cardano.Crypto.Hash as Crypto
+import qualified Cardano.Crypto.Hashing as Byron.Crypto
+import           Ouroboros.Consensus.Byron.Ledger.Block (ByronBlock)
+import           Ouroboros.Consensus.Byron.Ledger.Mempool (TxId (..))
+import           Ouroboros.Consensus.HardFork.Combinator
+import           Ouroboros.Consensus.Ledger.SupportsMempool (GenTx)
+import           Ouroboros.Consensus.Mock.Ledger.Block (SimpleBlock, TxId (..))
+import           Ouroboros.Consensus.Shelley.Ledger.Block (ShelleyBlock)
+import           Ouroboros.Consensus.Shelley.Ledger.Mempool (TxId (..))
+import           Ouroboros.Consensus.TypeFamilyWrappers
+import qualified Shelley.Spec.Ledger.TxData as Shelley
+
+-- | Convert a transaction ID to raw bytes.
+class ConvertTxId blk where
+  txIdToRawBytes :: TxId (GenTx blk) -> ByteString
+
+instance ConvertTxId ByronBlock where
+  txIdToRawBytes (ByronTxId txId) = Byron.Crypto.abstractHashToBytes txId
+  txIdToRawBytes (ByronDlgId dlgId) = Byron.Crypto.abstractHashToBytes dlgId
+  txIdToRawBytes (ByronUpdateProposalId upId) =
+    Byron.Crypto.abstractHashToBytes upId
```

Comment:
Is your editor auto-wrapping after `80` columns?  The stylish haskell rules allow up to `100`

Reply:
I just stick with 80 personally, but I can keep it to 100.